example/examples/echo-server.rs

```diff
@@ -23,17 +23,22 @@ extern crate futures;
 extern crate libp2p_secio as secio;
 extern crate libp2p_swarm as swarm;
 extern crate libp2p_tcp_transport as tcp;
+extern crate multiplex;
 extern crate tokio_core;
 extern crate tokio_io;
 
 use futures::future::{Future, IntoFuture, loop_fn, Loop};
 use futures::{Stream, Sink};
+use std::env;
 use swarm::{Transport, SimpleProtocol};
 use tcp::TcpConfig;
 use tokio_core::reactor::Core;
 use tokio_io::codec::length_delimited;
 
 fn main() {
+    // Determine which address to listen to.
+    let listen_addr = env::args().nth(1).unwrap_or("/ip4/0.0.0.0/tcp/10333".to_owned());
```

Comment:
It'd be nice to do some kind of assert so that providing more than 1 argument explodes instead of silently doing something unexpected but it's just an example so I'm not too bothered.

Reply:
I don't really like that change because it would make the example more complex to read (and the point of an example is to be easy to read).

Reply:
That's fair.